### PR TITLE
SuperTest: Fixes return type of host method

### PR DESCRIPTION
**Pull request by lourd**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test supertest`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [source code](https://github.com/visionmedia/supertest/blob/a70c1a09c9b4ee47b77b6b3f32f77fc7eee24107/lib/agent.js#L39-L42) and [relevant source unit test](https://github.com/visionmedia/supertest/blob/a70c1a09c9b4ee47b77b6b3f32f77fc7eee24107/test/supertest.js#L975-L977).

The `host` method of the `supertest` library has had an incorrect definition of the `host` method available. It's been incorrectly available on the default export

```ts
import request from 'supertest'
import express from 'express'

const app = express()
request(app).host('foo.com') // does not actually exist
request.agent(app).host('foo.com') // only exists when using `agent` API
```

And the return type has been incorrect, indicating that a request was being returned instead of the `SuperTest` object.

This PR fixes both of those problems.

### CODE REVIEW

**Comment by lourd** [comment]
The part of the tests that are failing don't have anything to do with my change:

```
The declaration doesn't match the JavaScript module 'supertest'. Reason: The JavaScript module exports a property named 'Test', which is missing from the declaration module.
```

The `supertest` module has exported a symbol named `Test` for almost a decade now, it's nothing new. I looked through the [source for `Test`](https://github.com/visionmedia/supertest/blob/master/lib/test.js#L15) to see how hard it would be, and it would be pretty significant — it would require changing/fixing the types for `superagent`, too. The `superagent` types show `Request` as an interface, but it's actually a class that `Test` extends.

Hopefully y'all understand and that failure won't block getting my change merged in 🙏

File: types/supertest/tslint.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
         "ban-types": false,
         "dt-header": false,
         "no-empty-interface": false,
+        "no-outside-dependencies": false,
```
Comment (lourd) [inline types/supertest/tslint.json:7]: Not sure why this is needed, the supertest types don't import from `buffer`. The only direction I could find in resolving it was https://github.com/DefinitelyTyped/DefinitelyTyped/issues/49372.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Waiting on 'TODO'

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## supertest ([<kbd>unpkg</kbd>](https://unpkg.com/browse/supertest@latest/))
was missing the following properties:
1. Test
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 24b235314c4c66cbceae9cb9be0d8b706cb36d59
</p>

**Comment by lourd** [comment]
I misunderstood the warning from danger in the CI logs as an error. There's no need to define any new symbols, I've deleted the todo.

**Review by RyanCavanaugh — APPROVED** [review]

**Pull request merged**